### DISCUSSION TITLE
I am including some images required by the program in the pyinstaller command using the --add-data option, however, the program is unable to find them.

### DISCUSSION THREAD

**Discussion by TechPro424** [pyinstaller/pyinstaller#8902] [Help] [answered]
`pyinstaller --add-data 'Images/*.png:Images'  --onefile --windowed 'Uno v2/starting_page.py'`

In my program, I'm trying to open an image which is in the path `<working directory>\\Images\\smth.png`
Works fine when I run the program normally.
However, when I try to run the exe generated by PyInstaller, I get the error
`Image import error - '[Errno 2] No such file or directory: '<working directory>\\Images\\green6.png''`

I have included the images in the pyinstaller command using the `--add-data` option.
Where am I going wrong?

**Comment by rokm** [comment, marked as answer]
>  Where am I going wrong?

Your images are not in the current working directory. See https://pyinstaller.org/en/stable/runtime-information.html#using-file

**Reply by TechPro424** [reply]
The images are in the current working directory when I run the program normally. I don't want them to be in the current working directory in case of the exe, but instead be bundled within the exe itself.

**Reply by rokm** [reply]
How are you referring to those images?

**Reply by TechPro424** [reply]
Is that not what the `dest` parameter is for in `--add-data '<source>:<dest>` is for? Or am I doing this wrong?

**Reply by TechPro424** [reply]
> How are you referring to those images?

`current_card_button = PushButton(app, height=3,width=4, image=logic.get_card_image(current[0]))`
`logic.get_card_image()` returns a path of the form `'Images\\image.png'`

**Reply by rokm** [reply]
> Is that not what the `dest` parameter is for in `--add-data '<source>:<dest>` is for? Or am I doing this wrong?

It is. 

But is also matters how you are trying to load those images *in your code*. Are you using relative paths (which implicitly assumes that the path is relative to current working directory)? Or are you anchoring them to `__file__` of the module that's trying to load the image, as described in the link that I gave you earlier?

**Reply by rokm** [reply]
> > How are you referring to those images?
> 
> `current_card_button = PushButton(app, height=3,width=4, image=logic.get_card_image(current[0]))` `logic.get_card_image()` returns a path of the form `'Images\\image.png'`

This assumes that `Images\image.png` exists in the current working directory. Which is not the case with the frozen application (nor with your python script, if you run it from an arbitrary directory).

**Reply by TechPro424** [reply]
The provided method works, thank you so much!
